bauhaus/core.py

```diff
@@ -380,6 +380,17 @@ class A(object):
 
     def wrapper(cls):
 
+        assert "_prop_name" in dir(cls), "Error: _prop_name must be defined in order for bauhaus to construct __repr__, __hash__, and __eq__"
+
+        def _init(self, val):
```

Comment:
Sorry I didn't catch this! I don't think the init should be defined. Which also means the test-core stuff that looks to self.val shouldn't work, unless an init like this is defined (I think you can do that with a base class over in the testcore file).

Many propositions will have >1 parameters, and the `._prop_name()` function will use them all to display things.

Reply:
To clarify, it is up to the user to define the __init__? I can remove it.

Reply:
This change does functionally change one of the tests, as written originally test_andor would fail as C() == C(). So I will still add in a value to enforce uniqueness as that seems like the intention

Reply:
Ya, no need to force the user to define `__init__`, but don't define it yourself in the wrapper. I could see the test-core stuff that would break, ya. Ok to rework them given the new structure of things (it's kind of a breaking change).

Reply:
The latest push should fix that. Tests pass and init is not defined in the wrapper.